docs/branches.md

```diff
@@ -41,6 +40,59 @@ branch `main`. If the local target had also moved compared to `main@origin`
 merged. If one is ahead of the other, then that target will be the new target.
 Otherwise, the local branch will be conflicted (see next section for details).
 
+As of December 2023 Jujutsu tracks[^1] and fetches all branches by default, 
+which is confusing users coming from Git. To smoothen the transition branch 
+tracking was introduced. 
+
+### What does `git.auto-local-branches` actually do? 
+
+Jujutsu fetches all Git Refs under `origin/refs/branches` by default and then 
+creates local branches for them. This is similar to Mercurial, which fetches 
+all it's Booksmarks (equivalent to Git branches) by default. If you toggle the 
```

Comment:
I'd make a new paragraph and reword slightly:

> You can disable this behavior with
>
> ```
> git.auto-local-branch=false
> ```
>
> Then, Jujutsu will only create local branches for remote branches which 
you explicitly track with `jj track`.

Reply:
I realized that we should probably mention `jj branch untrack` here as well. Something like:

> There are two ways to disable the creation (or modification) of the local branches by `jj git fetch`:
>
> - You can use `jj branch untrack` to stop tracking specific branches when fetching from specific remotes.
> - You can set `git.auto-local-branch=false` to change the default behavior. Then, Jujutsu will only create local branches for remote branches which you explicitly track with `jj track`.

Reply:
@yuja Please double-check that what I suggested is factually correct (in addition to any other suggestions you might have, of course).

Reply:
> > * You can set `git.auto-local-branch=false`

This will be the default, so we'll need to describe things in the other way around.

Reply:
We should probably do this in the commit that flips the default.

I'm still worried about `jj co main` not working, but I think it's worth a try to flip the default after a release. We can then see how many people complain and whether something needs to be adjusted before the following release.

Reply:
It's already done in the main branch. Regarding `jj co main`, `jj git clone` will automatically track the default remote branch no matter if `git.auto-local-branch` is off.

Reply:
> I realized that we should probably mention `jj branch untrack` here as well. Something like:
> 
> > There are two ways to disable the creation (or modification) of the local branches by `jj git fetch`:
> > 
> > * You can use `jj branch untrack` to stop tracking specific branches when fetching from specific remotes.
> > * You can set `git.auto-local-branch=false` to change the default behavior. Then, Jujutsu will only create local branches for remote branches which you explicitly track with `jj track`.

> > > * You can set `git.auto-local-branch=false`
> 
> This will be the default, so we'll need to describe things in the other way around.

Adressing both of these issues, the default has changed since ~3 weeks to `git.auto-local-branch = false`. I literally spent 30 minutes on how to rewrite the section for the inverted condition yesterday but couldn't find a passing rephrasing before updating it to the latest patch. 

But as Yuya noted in #2714, `branches.md` will need a major rewrite anyway and if it happens in a timely matter, this rewrite could also happen then. 

And everything under `docs/` which references automatic branch creation must be rewritten before the release.

Reply:
Rewritten to ilya's suggestion, I'm kinda tempted to just add a `TODO: correct this to the new default` above the controversial part above.

Reply:
> It's already done in the main branch. 

Oh! I didn't notice. That's probably a very good sign.

It seems that my suggestion was plain wrong then. IIUC (I haven't checked yet), the default is now already `false`. The docs should match the state of the prerelease version. (I'm just stating this explicitly, I think we're all agreed on this already).

> I'm kinda tempted to just add a TODO: correct this to the new default above the controversial part.

I'm OK with that if you want to be done with this PR ASAP. Having some mention of the new default in a TODO is better than no mention of it. Just make sure to say what the default changed to in the TODO. I'll hopefully then update the text in a separate PR. Or, if this doesn't get merged soon, I'll try to post a suggestion here once I come up with it.

I'm not sure whether @martinvonz is creating a release tomorrow. I think it'd be fine to skip this month, and it's also fine to keep to the schedule.

Reply:
Yes, I plan to create a release tomorrow as usual.

Reply:
I've added the TODO for the next person.